src/storage/src/render/persist_sink.rs

```diff
@@ -195,99 +162,102 @@ pub fn render<G>(
                         }
                     }
                 }
+                Event::Progress(input_upper) => {
```

Comment:
this reminds me, does timely periodically send an extra activation for just frontier updates, where no data will be in the input buffers? of could this branch be starved?

Reply:
It is [configurable](https://docs.rs/timely/latest/timely/dataflow/operators/generic/builder_rc/struct.OperatorBuilder.html#method.set_notify) and [true by default](https://github.com/TimelyDataflow/timely-dataflow/blob/master/timely/src/dataflow/operators/generic/builder_raw.rs#L37)